lib/path_funcs.sh

```diff
@@ -2,40 +2,40 @@ function platform_tools_path() {
   echo "${build_path}/.platform_tools"
 }
 
-function erlang_path() {
-  echo "$(platform_tools_path)/erlang"
-}
-
 function runtime_platform_tools_path() {
   echo "${runtime_path}/.platform_tools"
 }
 
+function build_erlang_path() {
+  echo "$(platform_tools_path)/erlang"
+}
+
 function runtime_erlang_path() {
   echo "$(runtime_platform_tools_path)/erlang"
 }
 
-function elixir_path() {
+function build_elixir_path() {
   echo "$(platform_tools_path)/elixir"
 }
 
-function generate_tmp_erlang_build_dir() {
-  # Do not call this in a subshell e.g. $(generate_tmp_erlang_build_dir)
-  tmp_erlang_build_dir="$(mktemp -d)"
+function runtime_elixir_path() {
+  echo "$(runtime_platform_tools_path)/elixir"
 }
 
-# We just go head and call it here before erlang_build_path is even defined so we should be good
-generate_tmp_erlang_build_dir
+function build_hex_home_path() {
+  echo "${build_path}/.hex"
+}
+
+function runtime_hex_home_path() {
+  echo "${runtime_path}/.hex"
+}
+
+function build_mix_home_path() {
+  echo "${build_path}/.mix"
+}
 
-function erlang_build_path() {
```

Comment:
since this was only used in one place, i moved it to a `local` variable in `install_erlang` to remove confusion and having multiple functions named after the erlang build path.